paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlTypeUtils.java

```diff
@@ -140,28 +139,37 @@ public class MySqlTypeUtils {
     private static final String RIGHT_BRACKETS = ")";
     private static final String COMMA = ",";
 
-    private static final List<String> HAVE_SCALE_LIST =
-            Arrays.asList(DECIMAL, NUMERIC, DOUBLE, REAL, FIXED);
-    private static final List<String> MAP_TO_DECIMAL_TYPES =
-            Arrays.asList(
-                    NUMERIC,
-                    NUMERIC_UNSIGNED,
-                    NUMERIC_UNSIGNED_ZEROFILL,
-                    FIXED,
-                    FIXED_UNSIGNED,
-                    FIXED_UNSIGNED_ZEROFILL,
-                    DECIMAL,
-                    DECIMAL_UNSIGNED,
-                    DECIMAL_UNSIGNED_ZEROFILL);
-
     private static final ObjectMapper objectMapper = new ObjectMapper();
 
-    public static DataType toDataType(String mysqlFullType, TypeMapping typeMapping) {
-        return toDataType(
-                MySqlTypeUtils.getShortType(mysqlFullType),
-                MySqlTypeUtils.getPrecision(mysqlFullType),
-                MySqlTypeUtils.getScale(mysqlFullType),
-                typeMapping);
+    public static Tuple3<String, Integer, Integer> getTypeInfo(String typeName) {
+        int leftBracketIndex = typeName.indexOf(LEFT_BRACKETS);
+        String shortType;
+        int length = 0;
+        int scale = 0;
+        if (leftBracketIndex != -1) {
+            int rightBracketIndex = typeName.indexOf(RIGHT_BRACKETS);
+            shortType =
+                    typeName.substring(0, leftBracketIndex).trim().toUpperCase()
+                            + typeName.substring(rightBracketIndex + 1).toUpperCase();
+
+            String insideBrackets =
+                    typeName.substring(leftBracketIndex + 1, rightBracketIndex).trim();
+            int commaIndex = insideBrackets.indexOf(COMMA);
+
+            if (commaIndex != -1 && !isEnumType(shortType) && !isSetType(shortType)) {
+                length = Integer.parseInt(insideBrackets.substring(0, commaIndex).trim());
+                scale = Integer.parseInt(insideBrackets.substring(commaIndex + 1).trim());
+            } else if (!isEnumType(shortType) && !isSetType(shortType)) {
+                length = Integer.parseInt(insideBrackets);
+            }
+        } else {
+            shortType = typeName.toUpperCase();
+            if (isDecimalType(shortType)) {
+                length = 38;
+                scale = 18;
```

Comment:
I think it's better to move comments to here:
```
// when missing precision of the decimal, we
// use the max precision to avoid parse error

// When missing scale of the decimal, we
// use the max scale to avoid parse error
```

Reply:
Thanks, done